include/ck_tile/core/arch/arch.hpp

```diff
@@ -60,13 +61,253 @@ enum struct memory_operation_enum : std::uint16_t
     add
 };
 
-CK_TILE_HOST_DEVICE constexpr index_t get_warp_size()
+/*! @enum amdgcn_target_arch_id
+ * @brief Defines constants for AMDGCN architecture target IDs
+ */
+enum struct amdgcn_target_arch_id
```

Comment:
one possible design choice here is to have a family of empty structs instead of enum

https://www.fluentcpp.com/2018/05/01/when-to-use-enums-and-when-to-use-tag-dispatching-in-cpp/

doing tag dispatch via structs allows to keep code for different architectures separate